### ISSUE TITLE
`get_argument_from_call` doesn't account for **kwargs

### ISSUE THREAD

**Issue by jacobtylerwalls** [pylint-dev/pylint#7761]
### Bug description

`get_argument_from_call` doesn't account for keyword arguments provided by unpacked kwargs. This is causing both false positives and false negatives

False negative
```python
kw={'mode': 'garbage'}
open(fp, **kw)
```

False positive
```python
kw={'maxsplit', 1}
'asdf'.split('s', **kw)[-1]
```

### Command used

```shell
pylint c.py
```

### Pylint output

first example: nothing

### Expected behavior

first example: Expected a warning for `bad-open-mode`

### Pylint version

```shell
pylint 2.16.0-dev
astroid 2.13.0-dev0
Python 3.11.0 (v3.11.0:deaf509e8f, Oct 24 2022, 14:43:23) [Clang 13.0.0 (clang-1300.0.29.30)]
```

**Comment by LeetaoGoooo** [comment]
I'd like to work on this if there are nobody else~

**Comment by jacobtylerwalls** [comment]
Terrific. Thanks for volunteering.

**Comment by jacobtylerwalls** [comment]
Hi @LeetaoGoooo, just an update. #8728 did a good portion of this issue, so I'll just pick up from here and look at the remaining false negatives. Let us know if you want any direction about something else to help out with!

**Comment by LeetaoGoooo** [comment]
> Hi @LeetaoGoooo, just an update. #8728 did a good portion of this issue, so I'll just pick up from here and look at the remaining false negatives. Let us know if you want any direction about something else to help out with!

Thank you for letting me know about the progress on the issue. I'm glad to hear that #8728 was able to address a good portion of it.  If there's anything else you think I can help with, please don't hesitate to let me know.
